ouroboros-consensus/src/Ouroboros/Consensus/Node/DbLock.hs

```diff
@@ -0,0 +1,67 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Ouroboros.Consensus.Node.DbLock (
+    DbLocked (..)
+  , dbLockFile
+  , lockDB
+  ) where
+
+import           Control.Concurrent
+import           Control.Monad (void)
+import           Data.Text (Text)
+import           Ouroboros.Consensus.Storage.FS.API
+import           Ouroboros.Consensus.Storage.FS.API.Types
+import           Ouroboros.Consensus.Util.ResourceRegistry
+
+import           Control.Exception
+import           System.FileLock
+
+-- | This can run from multiple processes concurrently, without any locking.
+-- It creates the db directory and its lock file, if any of them doesn't exist
+-- already.
+createDBFolder :: Monad m => HasFS m h -> m ()
+createDBFolder hasFS = do
+    createDirectoryIfMissing hasFS True root
+    void $ hOpen hasFS pFile (WriteMode AllowExisting)
```

Comment:
Quoting what you quoted in https://github.com/input-output-hk/ouroboros-network/issues/1903#issuecomment-609634186:
> [..], on Windows 'System.IO.openFile' for a lock file will fail when the lock is held

So this is wrong as it tries to open the lock file even if it's locked.

Question: does `tryLockFile` work when the file doesn't exist yet? If so, then that would be easier. Ah, see https://github.com/takano-akio/filelock/blob/9681ff960d2695cd67fe749edab0e2af06c6f829/System/FileLock/Internal/LockFileEx.hsc#L40

Reply:
No, it's the content of the file that is locked, not the file as a whole. That's why the failure in the logs ocured on reading the file, not opening the file.